src/generators/utils/pagingOperations.ts

```diff
@@ -0,0 +1,378 @@
+        NameType.Operation
+      );
+
+      const nextOperation = operationGroupDetails.operations.find(
+        o =>
+          o.name.toLocaleLowerCase() ===
+          operation.pagination?.nextLinkOperationName?.toLocaleLowerCase()
+      );
+
+      // We need to figure out the body response type to be able to set the return types properly
+      const bodyResponseType = getPagingResponseBodyType(operation);
+
+      // Since we'll be using this type name for importing, we need to make sure that it is not in
+      // the form of array [], just need the name.
+      let bodyResponseTypeName =
+        bodyResponseType?.typeName.replace("[]", "") || "";
+
+      // In case the type name collides with the operation group class we need to append Model to the name
+      if (bodyResponseTypeName === operationGroupDetails.name) {
+        bodyResponseTypeName = `${bodyResponseTypeName}Model`;
+      }
+
+      if (!bodyResponseType) {
+        throw new Error(
+          `Pageable operation ${initialOperationName} has no return values`
+        );
+      }
+
+      // Keep track of the models we'll need to import
+      bodyResponseType?.usedModels.forEach(m => importedModels.add(m));
+
+      let nextMethodParameters: MethodParameter[] | null = null;
+
+      if (nextOperation) {
+        nextMethodParameters = getOperationParameterSignatures(
+          nextOperation,
+          clientDetails.parameters,
+          importedModels,
+          operationGroupClass
+        ).baseMethodParameters.map(parameter => {
+          if (parameter.name === "nextLink") {
+            return { ...parameter, hasQuestionToken: true };
+          }
+
+          return parameter;
+        });
+      }
+
+      const {
+        baseMethodParameters: initialMethodParameters
+      } = getOperationParameterSignatures(
+        operation,
+        clientDetails.parameters,
+        importedModels,
+        operationGroupClass
+      );
+
+      // Build an object with all the information about the paging methods
+      // while generating each of the paging methods, this will help up access
+      // information about the other methdos.
+      const pagingMethodSettings: PagingMethodSettings = {
+        initialMethod: {
+          name: `${operation.namePrefix}${initialOperationName}`,
+          parameters: initialMethodParameters
+        },
+        nextMethod: nextMethodParameters
+          ? {
+              name: `${operation.namePrefix}${nextOperationName}`,
+              parameters: nextMethodParameters
+            }
+          : undefined,
+        publicMethod: {
+          name: getPublicMethodName(initialOperationName),
+          parameters: initialMethodParameters
+        },
+        allMethod: {
+          name: `${initialOperationName}PagingAll`,
+          parameters: initialMethodParameters
+        },
+        pageMethod: {
+          name: `${initialOperationName}PagingPage`,
```

Comment:
interesting name...

Reply:
I was giving it only Page but it was running into some naming conflicts, so I wanted a more unique name